### ISSUE TITLE
Feature Request: Expand upon the Multiplayer Side of Loco

### ISSUE THREAD

**Issue by Jump-Suit** [OpenRCT2/OpenLoco#95]
I propose the p2p system for the original game multiplayer be swapped with a more up-to-date system of Server Lists similar to that of OpenTTD's style. Granting ease of access to more profound multiplayer experiences in a flash!

**Comment by AaronVanGeffen** [comment]
Something like this, or indeed like OpenRCT2's, has indeed been on the wish list. Don't expect it too soon, though.

**Comment by IntelOrca** [comment]
A server list is trivial to add. Extending it to more than two players, not so much - it could require a lot of stuff to be implemented but it depends.

**Comment by jgottula** [comment]
> Extending it to more than two players, not so much - it could require a lot of stuff to be implemented but it depends.

Yeah, this is an area where *in principle* the game shouldn't have any particular problem with the idea of having more than two player-controlled companies at once; but you never know how many "only 2 non-AI companies will ever exist" assumptions we'll end up coming across in the code, given that it only ever took the form of a 2-player multiplayer game in practice.

**Comment by YetiWiz** [comment]
A cooperative mode would be nice, where companies could share and even modify each others tracks and stations. Also the ability to mark spots on the map.

**Comment by Jump-Suit** [comment]
I like the idea YetiWiz, in addition to what you said. I am also thinking perhaps have the player sharing your holdings be capable of limited capabilities from the main CEO. Similar to that of a Hierarchy.  CEO can enable/disable capabilities from the other player/players.

**Comment by Dracheron** [comment]
Heads up there's already a singleplayer/two player mode button that can be targeted to trigger the multiplayer system.  It's the bottom button in the main menu UI.

**Comment by Dracheron** [comment]
Heads up I just did my own investigation on the divers scenarios vs. their map sizes, and from what I could gather, unless the scenario itself was abnormally small, there was no correlation between the map size and the maximum number of companies that could compete.  Between this issue and possibly re-balancing the economy of the game and expanding map limits (issue #189 ), I don't see why we can simply ignore the AI and bypass the maximum number of companies allowed on a particular scenario, especially once sandbox mode is implemented.

Also, speaking of expanded tile sizes (which should be posted in #189 ), an interesting project that would be able to test how well the larger map size would do is to manually make a map of continental North America (Alaska, Canada, and Greenland and Iceland all the way down to Mexico and Northern Caribbean) combining the data from the divers U.S. maps as well as add a few cities that have risen to prominence since the game's release.

**Comment by Dracheron** [comment]
So... anyone got any idea as to test how AI companies work in a multiplayer setting beyond having two people play through a scenario vs a single person playing the same scenario?

**Comment by AaronVanGeffen** [comment]
The multiplayer part of OpenLoco is completely broken due to DirectPlay not being imported. As is noted in the release notes, you cannot use the multiplayer right now, unfortunately.

**Comment by Dracheron** [comment]
hmm... interesting... thats good to know, was considering such tests in the original base game since that would be the baseline to work with

**Comment by olivatooo** [comment]
I really hope to see multiplayer in the roadmap, love the project
